libs/langchain/langchain/document_loaders/pdf.py

```diff
@@ -440,3 +468,60 @@ def load(self) -> List[Document]:
         parser = PDFPlumberParser(text_kwargs=self.text_kwargs)
         blob = Blob.from_path(self.file_path)
         return parser.parse(blob)
+
+
+class AmazonTextractPDFLoader(BasePDFLoader):
+    """Loads a PDF from local file system or HTTP or S3 endpoints
+    and passes down to the AmazonTextractPDFParser to generate Documents
+        boto3_textract_client can be passed, for example when configuration is done in code and not through environment or when Textract should be called in a different region
+
+    """
+    try:
+        import textractcaller as tc
+    except ImportError:
+        raise ModuleNotFoundError(
+            "Could not import amazon-textract-caller python package. "
+            "Please install it with `pip install amazon-textract-caller`."
+        )
+
+    def __init__(self, file_path: str, textract_features:List[tc.Textract_Features]=list(), boto3_textract_client=None) -> None:
```

Comment:
Would prefer to use a shorter name for the boto3 client, may be `client`. Also, is it better to encapsulate the creation of the client inside the class rather than having users create it; make the client optional, so that if clients don't pass the boto3 client, we create one in the constructor.